consoleme/handlers/v2/typeahead.py

```diff
@@ -97,3 +100,58 @@ async def get(self):
             self.write(json.dumps([{"title": arn} for arn in arn_array.__root__]))
         else:
             self.write(arn_array.json())
+
+
+class SelfServiceStep1ResourceTypeahead(BaseAPIV2Handler):
+    async def get(self):
+        try:
+            type_ahead: Optional[str] = (
+                self.request.arguments.get("typeahead")[0].decode("utf-8").lower()
+            )
+        except TypeError:
+            type_ahead = None
+        if not type_ahead:
+            self.write(json.dumps([]))
+            return
+        try:
+            limit: int = self.request.arguments.get("limit")[0].decode("utf-8")
+            if limit:
+                limit = int(limit)
+        except TypeError:
+            limit = 20
+
+        typehead_data = await retrieve_json_data_from_redis_or_s3(
+            redis_key=config.get(
+                "cache_self_service_typeahead.redis.key",
+                "cache_self_service_typeahead_v1",
+            ),
+            s3_bucket=config.get("cache_self_service_typeahead.s3.bucket"),
+            s3_key=config.get(
+                "cache_self_service_typeahead.s3.file",
+                "cache_self_service_typeahead/cache_self_service_typeahead_v1.json.gz",
+            ),
+        )
+        matching = []
+
+        for entry in typehead_data.get("typeahead_entries", []):
+            if len(matching) >= limit:
```

Comment:
Do we have an upper bound on `limit`? If not, this could get costly for large values of `limit`. This could be avoided by (a) keeping a counter var or (b) setting a pretty conservative max for `limit`.

Reply:
Added max_limit with a sane default